kpm/lib/kpm/account.rb

```diff
@@ -0,0 +1,386 @@
+        end
+      end
+
+    end
+
+    def export_data(account_id = nil)
+
+      if account_id === :export.to_s
+        puts "\e[91;1mNeed to specify an account id\e[0m\n\n"
+        Dir.rmdir(TMP_DIR)
+        return
+      end
+
+      uri = URI("#{@killbill_url}/#{KILLBILL_API_VERSION}/kb/export/#{account_id}")
+      export_file = TMP_DIR + File::SEPARATOR + 'kbdump'
+
+      request = Net::HTTP::Get.new(uri.request_uri)
+      request.basic_auth(@killbill_user,@killbill_password)
+      request['X-Killbill-ApiKey'] = @killbill_api_key;
+      request['X-Killbill-ApiSecret'] = @killbill_api_secrets;
+      request['X-Killbill-CreatedBy'] = WHO;
+
+      response = Net::HTTP.start(uri.host,uri.port) do |http|
+        http.request(request)
+      end
+
+      if response.to_s.include? 'HTTPUnauthorized'
+        raise Interrupt, "User is unauthorized -> \e[93mUser[#{@killbill_user}],password[#{@killbill_password}],api_key[#{@killbill_api_key}],api_secret[#{@killbill_api_secrets}]"
+      end
+
+      open (export_file), 'w' do |io|
+        #io.write response.body
+        table_name = nil
+        cols_names = nil
+        response.body.split("\n").each do |line|
+          words = line.strip.split(" ")
+          clean_line = line
+
+          if /--/.match(words[0])
+
+            table_name = words[1]
+            cols_names = words[2].strip.split("|")
+
+          elsif not table_name.nil?
+
+            row = []
+            cols = line.strip.split("|")
+            cols_names.each_with_index { |col_name, index|
+              sanitized_value = sanitize_export(table_name,col_name,cols[index])
+
+              row << sanitized_value
+
+            }
+
+            clean_line = row.join("|")
+          end
+
+          io.puts clean_line
+        end
+
+      end if response.is_a?(Net::HTTPSuccess)
+
+      if not File.exist?(export_file)
+        puts "\e[91;1mAccount id not found\e[0m\n\n"
+        Dir.rmdir(TMP_DIR)
+      else
+        puts "\e[32mData exported under #{export_file}\e[0m\n\n"
+      end
+
+      export_file
+    end
+
+    def sanitize_export(table_name,col_name,value)
+
+      if not REMOVE_DATA_FROM[table_name.to_sym].nil?
+
+
+        if REMOVE_DATA_FROM[table_name.to_sym].include? col_name.to_sym
+          return nil
+        end
+
+      end
+
+      value
+    end
+
+    def import_data(source_file,account_record_id,tenant_record_id, skip_payment_methods)
+
+      @account_record_id = account_record_id
+      @tenant_record_id = tenant_record_id
+
+      if source_file === :import.to_s
+        puts "\e[91;1mNeed to specify a file\e[0m\n\n"
+        Dir.rmdir(TMP_DIR)
+        return
+      end
+
+      if not File.exist?(source_file)
+        puts "\e[91;1mNeed to specify a valid file\e[0m\n\n"
+        Dir.rmdir(TMP_DIR)
+        return
+      end
+
+      sanitize_and_import(source_file, skip_payment_methods)
+    end
+
+    def sanitize_and_import(source_file, skip_payment_methods)
```

Comment:
It would be nice if this method checks that the same number of rows that we had in the input for each table actually end up being inserted in the target table (maybe this already comes for free from the mysql import tool, unsure but something to check).